### PR TITLE
Fix SAMRecord.getReadLength() so it does not throw an exception if null bases

### PR DESCRIPTION
**Pull request by ronlevine**
### Description

Fixes https://github.com/samtools/htsjdk/issues/297.
Fixed SAMRecord.getReadLength by not calling getReadBases().length read bases is null.
Added a unit test.
Made private unit tests public so they can be run.

### Checklist

- [X] Code compiles correctly
- [X] New tests covering changes and new functionality
- [X] All tests passing
- [ ] Extended the README / documentation, if necessary
- [ ] Is not backward compatible (breaks binary or source compatibility)

### CODE REVIEW

**Comment by ronlevine** [comment]
@yfarjoun Please review.

File: src/test/java/htsjdk/samtools/SAMRecordUnitTest.java [hunk 1]
```diff
@@ -776,7 +776,7 @@ public void testNullHeaderRecordValidation() {
     }
 
     @Test
-    private void testNullHeaderDeepCopy() {
+    public void testNullHeaderDeepCopy() {
```
Comment (yfarjoun) [inline src/test/java/htsjdk/samtools/SAMRecordUnitTest.java]: oh snap, these were not being tested?

File: src/test/java/htsjdk/samtools/SAMRecordUnitTest.java [hunk 2]
```diff
@@ -1039,4 +1039,20 @@ public SAMRecord createTestSamRec() {
 
         return(rec);
     }
+
+    @DataProvider
+    public Object [][] readBasesGetReadLengthData() {
```
Comment (yfarjoun) [inline src/test/java/htsjdk/samtools/SAMRecordUnitTest.java]: Can you also add tests that set the bases using `setReadString`? in particular try to break it with an empty string and the "*" string that means the empty string for read bases.

File: src/test/java/htsjdk/samtools/SAMRecordUnitTest.java (hunk 1 above)
Comment (ronlevine) [inline src/test/java/htsjdk/samtools/SAMRecordUnitTest.java]: Yes. I assume snpp == &$%@.

File: src/test/java/htsjdk/samtools/SAMRecordUnitTest.java (hunk 2 above)
Comment (ronlevine) [inline src/test/java/htsjdk/samtools/SAMRecordUnitTest.java]: Done

**Comment by ronlevine** [comment]
@yfarjoun Responded to comments. Please take another look.
